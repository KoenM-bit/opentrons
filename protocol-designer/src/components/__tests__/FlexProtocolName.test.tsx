import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import { useFormikContext } from 'formik'
import {
  FlexProtocolNameComponent,
  flexProtocolName,
} from '../FlexProtocolEditor/FlexPillForm/FlexProtocolName'

jest.mock('formik', () => ({
  useFormikContext: jest.fn(),
}))

describe('FlexProtocolNameComponent', () => {
  const handleChange = jest.fn()
  const handleBlur = jest.fn()

  beforeEach(() => {
    ;(useFormikContext as jest.Mock).mockReturnValue({
      values: {
        fields: {
          name: '',
          author: '',
          description: '',
        },
      },
      errors: {},
      touched: {},
      handleChange,
      handleBlur,
    })
  })

  afterEach(() => {
    jest.resetAllMocks()
  })

  it('renders FlexProtocolNameComponent without errors/crashing', () => {
    render(<FlexProtocolNameComponent />)
  })

  it('renders protocol name input field', () => {
    const { container } = render(<FlexProtocolNameComponent />)
    const protocolNameInput = screen.getByTestId('protocol-name-input')

    expect(protocolNameInput).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('renders organization author input field', () => {
    const { container } = render(<FlexProtocolNameComponent />)
    const organizationAuthorInput = screen.getByTestId(
      'organization-author-input'
    )

    expect(organizationAuthorInput).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('renders protocol description textarea', () => {
    const { container } = render(<FlexProtocolNameComponent />)
    const protocolDescriptionTextarea = screen.getByTestId(
      'protocol-description-textarea'
    )

    expect(protocolDescriptionTextarea).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('renders protocol name input field with initial focus', () => {
    const { getByTestId, container } = render(<FlexProtocolNameComponent />)
    const protocolNameInput = getByTestId('protocol-name-input')

    expect(protocolNameInput).toHaveFocus()
    expect(container).toMatchSnapshot()
  })

  it('updates protocol name input field value', () => {
    const { values } = useFormikContext<flexProtocolName>()
    values.fields.name = 'New Protocol Name'
    const { container } = render(<FlexProtocolNameComponent />)
    const protocolNameInput = screen.getByTestId(
      'protocol-name-input'
    ) as HTMLInputElement

    fireEvent.change(protocolNameInput, {
      target: { value: 'New Protocol Name' },
    })

    expect(protocolNameInput.value).toBe('New Protocol Name')
    expect(container).toMatchSnapshot()
  })

  it('updates organization author input field value', () => {
    const { values } = useFormikContext<flexProtocolName>()
    values.fields.author = 'New Organization Author'

    const { container } = render(<FlexProtocolNameComponent />)
    const organizationAuthorInput = screen.getByTestId(
      'organization-author-input'
    ) as HTMLInputElement

    fireEvent.change(organizationAuthorInput, {
      target: { value: 'New Organization Author' },
    })

    expect(organizationAuthorInput.value).toBe('New Organization Author')
    expect(container).toMatchSnapshot()
  })

  it('updates protocol description textarea value', () => {
    const { values } = useFormikContext<flexProtocolName>()
    values.fields.description = 'New Protocol Description'
    const { container } = render(<FlexProtocolNameComponent />)
    const protocolDescriptionTextarea = screen.getByTestId(
      'protocol-description-textarea'
    ) as HTMLTextAreaElement

    fireEvent.change(protocolDescriptionTextarea, {
      target: { value: 'New Protocol Description' },
    })

    expect(protocolDescriptionTextarea.value).toBe('New Protocol Description')
    expect(container).toMatchSnapshot()
  })
})
