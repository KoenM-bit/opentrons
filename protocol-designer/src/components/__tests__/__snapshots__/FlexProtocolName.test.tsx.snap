// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexProtocolNameComponent renders organization author input field 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
}

.c1 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1rem;
}

.c2 {
  font-size: 0.813rem;
  font-weight: 400;
  line-height: 1rem;
}

.c3 {
  font-size: 0.625rem;
  font-weight: 400;
  line-height: 1rem;
}

<div>
  <h2
    class="c0 c1"
  >
    Protocol Name and Description
  </h2>
  <div
    class="flex_sub_heading"
  >
    <h5
      class="c0 c2"
    >
      Choose a name for your protocol.*
    </h5>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Protocol Name
    </p>
    <div
      class="form_field"
    >
      <div
        class="input_field_container"
      >
        <div
          class="input_field"
        >
          <input
            data-testid="protocol-name-input"
            id="protocol-name-input"
            name="fields.name"
            tabindex="1"
            type="text"
            value=""
          />
        </div>
        <div
          class="input_caption"
        >
          <span />
          <span
            class="right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="flex_sub_heading"
  >
    <h5
      class="c0 c2"
    >
      Add more information, if you like (you can change this later).
    </h5>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Organization/Author
    </p>
    <div
      class="form_field"
    >
      <div
        class="input_field_container"
      >
        <div
          class="input_field"
        >
          <input
            data-testid="organization-author-input"
            id="organization-author-input"
            name="fields.author"
            tabindex="2"
            type="text"
            value=""
          />
        </div>
        <div
          class="input_caption"
        >
          <span />
          <span
            class="right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Description
    </p>
    <textarea
      class="textarea_input"
      data-testid="protocol-description-textarea"
      name="fields.description"
      rows="4"
      tabindex="3"
    />
  </div>
</div>
`;

exports[`FlexProtocolNameComponent renders protocol description textarea 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
}

.c1 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1rem;
}

.c2 {
  font-size: 0.813rem;
  font-weight: 400;
  line-height: 1rem;
}

.c3 {
  font-size: 0.625rem;
  font-weight: 400;
  line-height: 1rem;
}

<div>
  <h2
    class="c0 c1"
  >
    Protocol Name and Description
  </h2>
  <div
    class="flex_sub_heading"
  >
    <h5
      class="c0 c2"
    >
      Choose a name for your protocol.*
    </h5>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Protocol Name
    </p>
    <div
      class="form_field"
    >
      <div
        class="input_field_container"
      >
        <div
          class="input_field"
        >
          <input
            data-testid="protocol-name-input"
            id="protocol-name-input"
            name="fields.name"
            tabindex="1"
            type="text"
            value=""
          />
        </div>
        <div
          class="input_caption"
        >
          <span />
          <span
            class="right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="flex_sub_heading"
  >
    <h5
      class="c0 c2"
    >
      Add more information, if you like (you can change this later).
    </h5>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Organization/Author
    </p>
    <div
      class="form_field"
    >
      <div
        class="input_field_container"
      >
        <div
          class="input_field"
        >
          <input
            data-testid="organization-author-input"
            id="organization-author-input"
            name="fields.author"
            tabindex="2"
            type="text"
            value=""
          />
        </div>
        <div
          class="input_caption"
        >
          <span />
          <span
            class="right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Description
    </p>
    <textarea
      class="textarea_input"
      data-testid="protocol-description-textarea"
      name="fields.description"
      rows="4"
      tabindex="3"
    />
  </div>
</div>
`;

exports[`FlexProtocolNameComponent renders protocol name input field 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
}

.c1 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1rem;
}

.c2 {
  font-size: 0.813rem;
  font-weight: 400;
  line-height: 1rem;
}

.c3 {
  font-size: 0.625rem;
  font-weight: 400;
  line-height: 1rem;
}

<div>
  <h2
    class="c0 c1"
  >
    Protocol Name and Description
  </h2>
  <div
    class="flex_sub_heading"
  >
    <h5
      class="c0 c2"
    >
      Choose a name for your protocol.*
    </h5>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Protocol Name
    </p>
    <div
      class="form_field"
    >
      <div
        class="input_field_container"
      >
        <div
          class="input_field"
        >
          <input
            data-testid="protocol-name-input"
            id="protocol-name-input"
            name="fields.name"
            tabindex="1"
            type="text"
            value=""
          />
        </div>
        <div
          class="input_caption"
        >
          <span />
          <span
            class="right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="flex_sub_heading"
  >
    <h5
      class="c0 c2"
    >
      Add more information, if you like (you can change this later).
    </h5>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Organization/Author
    </p>
    <div
      class="form_field"
    >
      <div
        class="input_field_container"
      >
        <div
          class="input_field"
        >
          <input
            data-testid="organization-author-input"
            id="organization-author-input"
            name="fields.author"
            tabindex="2"
            type="text"
            value=""
          />
        </div>
        <div
          class="input_caption"
        >
          <span />
          <span
            class="right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Description
    </p>
    <textarea
      class="textarea_input"
      data-testid="protocol-description-textarea"
      name="fields.description"
      rows="4"
      tabindex="3"
    />
  </div>
</div>
`;

exports[`FlexProtocolNameComponent renders protocol name input field with initial focus 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
}

.c1 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1rem;
}

.c2 {
  font-size: 0.813rem;
  font-weight: 400;
  line-height: 1rem;
}

.c3 {
  font-size: 0.625rem;
  font-weight: 400;
  line-height: 1rem;
}

<div>
  <h2
    class="c0 c1"
  >
    Protocol Name and Description
  </h2>
  <div
    class="flex_sub_heading"
  >
    <h5
      class="c0 c2"
    >
      Choose a name for your protocol.*
    </h5>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Protocol Name
    </p>
    <div
      class="form_field"
    >
      <div
        class="input_field_container"
      >
        <div
          class="input_field"
        >
          <input
            data-testid="protocol-name-input"
            id="protocol-name-input"
            name="fields.name"
            tabindex="1"
            type="text"
            value=""
          />
        </div>
        <div
          class="input_caption"
        >
          <span />
          <span
            class="right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="flex_sub_heading"
  >
    <h5
      class="c0 c2"
    >
      Add more information, if you like (you can change this later).
    </h5>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Organization/Author
    </p>
    <div
      class="form_field"
    >
      <div
        class="input_field_container"
      >
        <div
          class="input_field"
        >
          <input
            data-testid="organization-author-input"
            id="organization-author-input"
            name="fields.author"
            tabindex="2"
            type="text"
            value=""
          />
        </div>
        <div
          class="input_caption"
        >
          <span />
          <span
            class="right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Description
    </p>
    <textarea
      class="textarea_input"
      data-testid="protocol-description-textarea"
      name="fields.description"
      rows="4"
      tabindex="3"
    />
  </div>
</div>
`;

exports[`FlexProtocolNameComponent updates organization author input field value 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
}

.c1 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1rem;
}

.c2 {
  font-size: 0.813rem;
  font-weight: 400;
  line-height: 1rem;
}

.c3 {
  font-size: 0.625rem;
  font-weight: 400;
  line-height: 1rem;
}

<div>
  <h2
    class="c0 c1"
  >
    Protocol Name and Description
  </h2>
  <div
    class="flex_sub_heading"
  >
    <h5
      class="c0 c2"
    >
      Choose a name for your protocol.*
    </h5>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Protocol Name
    </p>
    <div
      class="form_field"
    >
      <div
        class="input_field_container"
      >
        <div
          class="input_field"
        >
          <input
            data-testid="protocol-name-input"
            id="protocol-name-input"
            name="fields.name"
            tabindex="1"
            type="text"
            value=""
          />
        </div>
        <div
          class="input_caption"
        >
          <span />
          <span
            class="right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="flex_sub_heading"
  >
    <h5
      class="c0 c2"
    >
      Add more information, if you like (you can change this later).
    </h5>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Organization/Author
    </p>
    <div
      class="form_field"
    >
      <div
        class="input_field_container"
      >
        <div
          class="input_field"
        >
          <input
            data-testid="organization-author-input"
            id="organization-author-input"
            name="fields.author"
            tabindex="2"
            type="text"
            value="New Organization Author"
          />
        </div>
        <div
          class="input_caption"
        >
          <span />
          <span
            class="right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Description
    </p>
    <textarea
      class="textarea_input"
      data-testid="protocol-description-textarea"
      name="fields.description"
      rows="4"
      tabindex="3"
    />
  </div>
</div>
`;

exports[`FlexProtocolNameComponent updates protocol description textarea value 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
}

.c1 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1rem;
}

.c2 {
  font-size: 0.813rem;
  font-weight: 400;
  line-height: 1rem;
}

.c3 {
  font-size: 0.625rem;
  font-weight: 400;
  line-height: 1rem;
}

<div>
  <h2
    class="c0 c1"
  >
    Protocol Name and Description
  </h2>
  <div
    class="flex_sub_heading"
  >
    <h5
      class="c0 c2"
    >
      Choose a name for your protocol.*
    </h5>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Protocol Name
    </p>
    <div
      class="form_field"
    >
      <div
        class="input_field_container"
      >
        <div
          class="input_field"
        >
          <input
            data-testid="protocol-name-input"
            id="protocol-name-input"
            name="fields.name"
            tabindex="1"
            type="text"
            value=""
          />
        </div>
        <div
          class="input_caption"
        >
          <span />
          <span
            class="right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="flex_sub_heading"
  >
    <h5
      class="c0 c2"
    >
      Add more information, if you like (you can change this later).
    </h5>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Organization/Author
    </p>
    <div
      class="form_field"
    >
      <div
        class="input_field_container"
      >
        <div
          class="input_field"
        >
          <input
            data-testid="organization-author-input"
            id="organization-author-input"
            name="fields.author"
            tabindex="2"
            type="text"
            value=""
          />
        </div>
        <div
          class="input_caption"
        >
          <span />
          <span
            class="right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Description
    </p>
    <textarea
      class="textarea_input"
      data-testid="protocol-description-textarea"
      name="fields.description"
      rows="4"
      tabindex="3"
    >
      New Protocol Description
    </textarea>
  </div>
</div>
`;

exports[`FlexProtocolNameComponent updates protocol name input field value 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
}

.c1 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1rem;
}

.c2 {
  font-size: 0.813rem;
  font-weight: 400;
  line-height: 1rem;
}

.c3 {
  font-size: 0.625rem;
  font-weight: 400;
  line-height: 1rem;
}

<div>
  <h2
    class="c0 c1"
  >
    Protocol Name and Description
  </h2>
  <div
    class="flex_sub_heading"
  >
    <h5
      class="c0 c2"
    >
      Choose a name for your protocol.*
    </h5>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Protocol Name
    </p>
    <div
      class="form_field"
    >
      <div
        class="input_field_container"
      >
        <div
          class="input_field"
        >
          <input
            data-testid="protocol-name-input"
            id="protocol-name-input"
            name="fields.name"
            tabindex="1"
            type="text"
            value="New Protocol Name"
          />
        </div>
        <div
          class="input_caption"
        >
          <span />
          <span
            class="right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="flex_sub_heading"
  >
    <h5
      class="c0 c2"
    >
      Add more information, if you like (you can change this later).
    </h5>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Organization/Author
    </p>
    <div
      class="form_field"
    >
      <div
        class="input_field_container"
      >
        <div
          class="input_field"
        >
          <input
            data-testid="organization-author-input"
            id="organization-author-input"
            name="fields.author"
            tabindex="2"
            type="text"
            value=""
          />
        </div>
        <div
          class="input_caption"
        >
          <span />
          <span
            class="right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="form_group"
  >
    <p
      class="c0 c3"
    >
      Description
    </p>
    <textarea
      class="textarea_input"
      data-testid="protocol-description-textarea"
      name="fields.description"
      rows="4"
      tabindex="3"
    />
  </div>
</div>
`;
